status/status.go

```diff
@@ -12,6 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// Package status contains constants for mesh and operation status.
 package status
```

Comment:
The presence of this package is a little disconcerting. Adapters are stateless currently. Ideally, they will stay this way. An adapter is not the source of truth for the status of a service mesh. I may not be thinking of the `status.go` package in the right context, though.  // @kumarabd

Reply:
> The presence of this package is a little disconcerting. Adapters are stateless currently. Ideally, they will stay this way. An adapter is not the source of truth for the status of a service mesh. I may not be thinking of the `status.go` package in the right context, though.  // @kumarabd 

This package is just a representation of the status strings as keys. They do not provide any stateful or stateless feature, that's upto where this is implemented.

Reply:
I think I remove `mesh and`. As pointed out, it's misleading and not intended.

Reply:
@kumarabd the Meshery adapter library will be implemented by Meshery adapters. Meshery adapters are stateless. We're creating constants to track the state of a service mesh, deployment, manifest that is being applied. 

MeshSync is the source of truth for the state of a given element (object) under management. Whether this state is written to disk or only kept in memory, unless these properties are transiently used only for the duration of a given operation, adapters begin to hold state. If doing so is needed to achieve a particular use case, let's chat about it so that we're clear that state exists in adapters and can be considered in the overall system design. State can be dangerous. The less locations we have for state, the better. Today, an adapter can come and go, be upgraded in-place without care. Deployments can have multiple adapters of the same type without concern for which adapter is executing a given operation. If we need to move away from this principle of adapter design, we'll need to pause to given holistic consideration as to how doing so affects the overall system architecture and operation.

Reply:
I have now updated the text. 

The way the status constants are used in the code is to indicate which one of the different phases during the lifetime of an operation is the current one. Status is used in error messages to indicate where/when an error occurred. From that point of view, I don't consider that to reflect state of the adapter, a deployment, or the mesh (the usage of this term was on oversight on my side). It is not used to determine whether a specific deployment has already been applied. Maybe it should not be called `status`, but `deploymentPhase`?  Or it could be implemented differently, for instance using different error codes.